src/connection.ts

```diff
@@ -3497,7 +3564,14 @@ Connection.prototype.STATE = {
                 }, callback);
               };
 
-              if (authentication.type === 'azure-active-directory-password') {
+              if (authentication.type === SupportedAuthenticationTypes.AzureActiveDirectoryAccessTokenGenerator) {
+                authentication
+                  .options
+                  .tokenGenerator
+                  .getToken()
+                  .then((token: string) => callback(null, token))
+                  .catch((err) => callback(err));
```

Comment:
I don't think we need to introduce a new authentication type for this. The `AzureActiveDirectoryAccessToken` can check whether the given `token` is a function, and if it is, do `await options.token.call(null)` inside the `sendLogin7Packet` method.

Does that make sense?

Reply:
That shouldn't be a part of this PR 🙈

Reply:
😬

Reply:
Don't worry. I'll let you finish this up first before leaving any more comments. 🙇‍♂️

Reply:
Sorry! That is PR only for the refactor. You saw my PoC of the new authentication type, but let's keep the discussion about that for the next PR :D